controllers/userController.js

```diff
@@ -0,0 +1,106 @@
+const User = require('../models/userModel')
+const generateToken = require('../utils/generateTokens');
+
+//@desc   signup new user instructor
+//@route  POST /api/register/instructor
+//@access Public
+const signupInstructor = async (req, res) => {
+    const { firstName, lastName, dob, email, password } = req.body;
+
+    const userExist = await User.findOne({ email });
+    if (userExist) {
+        res.status(400);
+        throw new Error('User already exists');
+    }
+
+    const instructor = await User.create({
+        firstName,
+        lastName,
+        dob,
+        email,
+        password,
+        type: 'instructor',
+    });
+    if (instructor) {
+        res.status(201).json({
```

Comment:
when I push my code ill let you call my function to populate the table with the bio specialisation and other fields